File: ExamenMoria_Java/src/main/java/models/item/Vara.java
package models.item;

import interfaces.Mago;

import java.util.UUID;

public class Vara extends Item implements Mago {
    private int poderVara; //Empieza en 0

    private int poderTotalDisponible; //Entre 1 y un valor parametrizable, por ejemplo 30

    public Vara(UUID id, int poderTotalVara) {
        super(id);
        this.poderVara = 0;
        this.poderTotalDisponible = poderTotalVara;
    }

    @Override
    public String toString() {
        return "Vara{" +
                "poderVara=" + poderVara +
                ", poderTotalDisponible=" + poderTotalDisponible +
                ", id=" + id +
                '}';
    }

    @Override
    public void recargarVara(int energia) {
        if (poderTotalDisponible == 0){
            System.out.println("Se intenta recargar los puntos de poder de la vara, pero como no queda energía para eso, no ocurre nada.");
            return;
        }
        if(energia <= poderTotalDisponible){
            poderVara = energia;
            poderTotalDisponible -= energia;
        }else{
            poderVara = poderTotalDisponible;
            poderTotalDisponible = 0;
        }
        System.out.println("Se recarga la vara con un total de "+energia+" puntos de energía.\nA la vara le queda un total de "+poderTotalDisponible+" puntos de poder disponibles.");
    }

    /**
     * Devuelve el poder actual de la vara devolviendolo a 0
     * @return el poder actual de la vara
     */
    @Override
    public Integer poderVara() {
        int poder = poderVara;
        poderVara = 0;
        return poder;
    }
}

File: ExamenMoria-Python/projecto/models/items/Vara.py
from Item import Item
from interfaz.Mago import Mago

class Vara(Item, Mago):

    poderVara = 0

    def __init__(self, id: int, poderTotalVara: int):
        super().__init__(id)
        self.poderTotalVara = poderTotalVara

    def __str__(self) -> str:
        return f"Vara({self.id}, {self.poderTotalVara}, {self.poderVara})"

    def recargarVara(self, energia: int):
        if self.poderTotalVara == 0:
            print("Se intenta recargar los puntos de poder de la vara, pero como no queda energía para eso, no ocurre nada.")
            return
        if self.poderTotalVara >= energia:
            self.poderVara = energia
            self.poderTotalVara -= energia
        else:
            self.poderVara = self.poderTotalVara
            self.poderTotalVara = 0
        print("Se recarga la vara con un total de "+str(energia)+" puntos de energía.\nA la vara le queda un total de "+str(self.poderTotalVara)+" puntos de poder disponibles.")


    def obtenerPoderVara(self):
        poder = self.poderVara
        self.poderVara = 0
        return poder